cmd/influxd/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,69 @@
+package upgrade
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/docker/docker/pkg/testutil/assert"
+	"github.com/influxdata/influxdb/v2/bolt"
+)
+
+func TestPathValidations(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	v1Dir := filepath.Join(tmpdir, "v1db")
+	v2Dir := filepath.Join(tmpdir, "v2db")
+
+	boltPath := filepath.Join(v2Dir, bolt.DefaultFilename)
+	enginePath := filepath.Join(v2Dir, "engine")
+
+	err = os.MkdirAll(filepath.Join(enginePath, "db"), 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	largs := make([]string, 0, 9)
+	largs = append(largs, "--username", "my-user")
+	largs = append(largs, "--password", "my-password")
+	largs = append(largs, "--org", "my-org")
+	largs = append(largs, "--bucket", "my-bucket")
+	largs = append(largs, "--retention", "7d")
+	largs = append(largs, "--token", "my-token")
+	largs = append(largs, "--v1-dir", v1Dir)
+	largs = append(largs, "--bolt-path", boltPath)
+	largs = append(largs, "--engine-path", enginePath)
+	largs = append(largs, "--config-file", "")
+
+	cmd := NewCommand()
+	cmd.SetArgs(largs)
+
+	if err := cmd.Execute(); err == nil {
+		t.Fatal("Must fail")
+	} else {
+		assert.Contains(t, err.Error(), "1.x metadb error")
+	}
```

Comment:
You can additionally import "github.com/stretchr/testify/require" and then use:

```suggestion
	err = cmd.Execute()
	require.Error(t, err)
	assert.Contains(t, err.Error(), "1.x metadb error")
```

Reply:
Well, I'm a bit confused from the InfluxDB tests code style. I've seen a lot of tests that prefer to use this than the require package. But, great, I personally like the require package. 👍

Reply:
@vlastahajek there is a difference between `require` and `assert`.

When a `require` condition fails, the test will abort. When an `assert` condition fails, it will continue.

It is clear here, that we _require_ an error and then following that we _assert_ the error is the correct value.

Reply:
Yes, I know. I use require in the influxdb-client-go tests heavy . I just felt that the server basic test style doesn't use require.